repos/system_upgrade/common/actors/peseventsscanner/libraries/peseventsscanner_repomap.py

```diff
@@ -0,0 +1,296 @@
+                                 equivalent exactly matching a source repository was not found.
+        :type default_channels: List[str]
+        :param prio_channel: Prefer repositories with this channel when looking for target equivalents.
+        :type prio_channel: str
+        """
+        # NOTE: currently I am keeping this default data structure that is not
+        # ideal for work, but there is not any significant impact..
+        self.repositories = repo_map.repositories
+        self.mapping = repo_map.mapping
+        # FIXME(pstodulk): what about default_channel -> fallback_channel
+        # hardcoded always as ga? instead of list of channels..
+        # it'd be possibly confusing naming now...
+        self.default_channels = default_channels or ['ga']
+        self.prio_channel = prio_channel
+
+    def set_default_channels(self, default_channels):
+        """
+        Set the default channels that are used as a fallback when searching
+        for the right target repository.
+
+        Usually it's not problem to find a target repository that matches
+        the source repository, however in some cases the target repository
+        doesn't have to be available in the required (premium) channel but
+        could be present in the standard one.
+
+        This is used usually for fallbacks and it's prerequisite for the time
+        the required channel will not be present for the particular repository.
+        E.g. can happen for layered products which has different lifecycles
+        and doesn't have to provide a special premium channels at all. E.g.
+        the Extras repository has GA and Beta channels, but no EUS. In case the
+        EUS is required, returns the GA one instead when this is present.
+
+        It's recommended to make the GA ('ga') always present in the default
+        list.
+
+        :param default_channels: Default channels to use when a target equivalent to a source repository that
+                                 matches its target channel properties exactly could not be found.
+        :type default_channels: List[str]
+        """
+        self.default_channels = default_channels
+
+    def get_pesid_repo_entry(self, repoid, major_version):
+        """
+        Retrieve the PESIDRepositoryEntry that matches the given repoid and OS major version.
+
+        :param repoid: RepoID that should the PESIDRepositoryEntry match.
+        :type repoid: str
+        :param major_version: RepoID that should the PESIDRepositoryEntry match.
+        :type major_version: str
+        :return: The PESIDRepositoryEntry matching the given repoid and major_version or None if no such
+                 entry could be found.
+        :rtype: Optional[PESIDRepositoryEntry]
+        """
+        for pesid_repo in self.repositories:
+            if pesid_repo.repoid == repoid and pesid_repo.major_version == major_version:
+                return pesid_repo
+        return None
```

Comment:
```suggestion
        return next(iter(self.get_pesid_repos(repoid, major_version)), None)
```
That's about the gist of this function, no?